blocklist-client/src/client/sanctions.rs

```diff
@@ -0,0 +1,308 @@
+//! This module interacts with the risk API to determine the risk severity associated with a user wallet address.
+//!
+//! It provides functionality to:
+//! - Check if given address is under sanctions
+//!
+//! The module includes functions to handle API requests, interpret responses, and map them to application-specific errors.
+
+use crate::common::error::Error;
+use crate::common::{BlocklistStatus, RiskAssessment, RiskSeverity};
+use crate::config::RiskAnalysisConfig;
+use reqwest::{Client, Response, StatusCode};
+use serde::Deserialize;
+use std::error::Error as StdError;
+use tracing::debug;
+const API_BASE_PATH: &str = "/api/v1/address";
+
+/// Represents the identification information for a blockchain address.
+#[derive(Debug, Deserialize)]
+pub struct Identification {
+    /// The Chainalysis Entity category. For sanctioned addresses, the value will be 'sanctions'.
+    #[serde(rename = "category")]
+    pub _category: String,
+    /// The OFAC name associated with the sanctioned address.
+    #[serde(rename = "name")]
+    pub _name: Option<String>,
+    /// The OFAC description of the sanctioned address.
+    #[serde(rename = "description")]
+    pub _description: Option<String>,
+    /// The OFAC URL for more information about the sanctioned address.
+    #[serde(rename = "url")]
+    pub _url: Option<String>,
+}
+
+/// Structure of responce of the sanctions API
+#[derive(Debug, Deserialize)]
+pub struct SanctionsResponce {
+    /// Array with sanctions data. If empty, the address is not blocklisted.
+    pub identifications: Vec<Identification>,
+}
+
+fn risk_assessment_path(base_url: &str, address: &str) -> String {
+    format!("{}{}/{}", base_url, API_BASE_PATH, address)
+}
+
+/// Check risk status associated with a registered address
+async fn get_risk_assessment(
+    client: &Client,
+    config: &RiskAnalysisConfig,
+    address: &str,
+) -> Result<RiskAssessment, Error> {
+    let api_url = risk_assessment_path(&config.api_url, address);
+    debug!("Beginning risk assessment for address: {address}");
+    let response = client
+        .get(&api_url)
+        .header("X-API-Key", &config.api_key)
+        .header("Accept", "application/json")
+        .send()
+        .await?;
+    let checked_response = check_api_response(response).await?;
+    let resp_result = checked_response.json::<SanctionsResponce>().await;
+    match resp_result {
+        Ok(resp) => {
+            if resp.identifications.is_empty() {
```

Comment:
Can you add a comment here explaining that if the address is sanctioned it maps it directly to `SEVERE` so the behavior is clear at a glance?